mp4/mdat_test.go

```diff
@@ -115,16 +114,14 @@ func TestReadData_LazyMdatMode(t *testing.T) {
 		t.Error(err)
 	}
 
-	// test ReadData with provided ReadSeeker
 	lazyMdat := &MdatBox{
 		StartPos:        0,
-		decLazyDataSize: 7,
-		mu:              &sync.Mutex{},
+		decLazyDataSize: 6,
```

Comment:
should this be `7`?

Reply:
The reason for not reading the full is that I by mistake first used the io.copy instead of io.copyN function, so I wanted to a test that makes a difference. Of course, that requires that the ReadData test is copied to SeekAndCopyData.

Maybe not the best name, but the way. CopyData should be better since I didn't make the seek mandatory.